sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/_tests_/e2e/student-scholastic-standings.institutions.controller.saveScholasticStanding.e2e-spec.ts

```diff
@@ -0,0 +1,321 @@
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeInstitutionLocation,
+  createFakeStudentAppeal,
+  getProviderInstanceForModule,
+  saveFakeApplicationDisbursements,
+} from "@sims/test-utils";
+import {
+  BEARER_AUTH_TYPE,
+  InstitutionTokenTypes,
+  authorizeUserTokenForLocation,
+  createTestingAppModule,
+  getAuthRelatedEntities,
+  getInstitutionToken,
+} from "../../../../testHelpers";
+import * as request from "supertest";
+import {
+  ApplicationStatus,
+  AssessmentTriggerType,
+  InstitutionLocation,
+  StudentAssessmentStatus,
+  StudentScholasticStandingChangeType,
+} from "@sims/sims-db";
+import {
+  APPLICATION_NOT_FOUND,
+  FormNames,
+  FormService,
+  INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+} from "../../../../services";
+import { APPLICATION_CHANGE_NOT_ELIGIBLE } from "../../../../constants";
+import { ScholasticStandingAPIInDTO } from "../../models/student-scholastic-standings.dto";
+import { addDays, getISODateOnlyString } from "@sims/utilities";
+import { AppInstitutionsModule } from "../../../../app.institutions.module";
+import { TestingModule } from "@nestjs/testing";
+
+describe("StudentScholasticStandingsInstitutionsController(e2e)-saveScholasticStanding.", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let db: E2EDataSources;
+  let collegeFLocation: InstitutionLocation;
+  let institutionUserToken: string;
+  const SCHOLASTIC_STANDING_FORM_NAME =
+    FormNames.ReportScholasticStandingChange;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+    // College F.
+    const { institution: collegeF } = await getAuthRelatedEntities(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    collegeFLocation = createFakeInstitutionLocation({ institution: collegeF });
+    await authorizeUserTokenForLocation(
+      db.dataSource,
+      InstitutionTokenTypes.CollegeFUser,
+      collegeFLocation,
+    );
+    // Institution token.
+    institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+  });
+
+  it("Should throw bad request exception error when the payload is invalid for formIO dryRun test.", async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(db.dataSource, {
+      institutionLocation: collegeFLocation,
+    });
+    const payload = {
+      data: {
+        booksAndSupplies: 1000,
+        scholasticStandingChangeType:
+          StudentScholasticStandingChangeType.ChangeInIntensity,
+      },
+    };
+    // Mock the form service to validate the dry-run submission result.
+    // TODO: Form service must be hosted for E2E tests to validate dry run submission
+    // and this mock must be removed.
+    const formService = await getProviderInstanceForModule(
```

Comment:
The result of this method can be centralized at the `beforeAll`.
It would also be good to have a beforeEach resetMocks in place.
```ts
beforeEach(async () => {
  jest.resetAllMocks();
});
```

Reply:
The suggestion from the comments could be something as below.
```ts
beforeEach(async () => {
  jest.resetAllMocks();
});
```
Create a centralized method to handle the form.io mock as below.
```ts
function mockFormioDryRun(options?: { validDryRun?: boolean }): void {
    const validDryRun = options?.validDryRun ?? true;
    payload = {
      data: {
        booksAndSupplies: 1000,
        dateOfChange: getISODateOnlyString(new Date()),
        scholasticStandingChangeType:
          StudentScholasticStandingChangeType.ChangeInIntensity,
      },
    };
    formService.dryRunSubmission = jest.fn().mockResolvedValue({
      valid: validDryRun,
      formName: FormNames.ReportScholasticStandingChange,
      data: { data: payload.data },
    });
  }
```

Reply:
The current approach works also. The suggested one can be open to accepting more changes and options.